### ISSUE TITLE
RedundantImportCheck does not detect specific and wildcard imports from the same package

### ISSUE THREAD

**Issue by Vantuz** [checkstyle/checkstyle#6450]
```
/var/tmp $ javac TestClass.java

/var/tmp $ cat checkstyle.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="RedundantImport"/>
    </module>
</module>


/var/tmp $ cat TestClass.java
import java.util.*;
import java.util.ArrayList;
import java.util.List;

public class TestClass {
    private List<String> list = new ArrayList<>();
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.17-all.jar -c checkstyle.xml TestClass.java
Starting audit...
Audit done.

```

---------------
Checkstyle reports no errors here

I expect Checkstyle to report violations:

- on line 2 that says that `java.util.ArrayList` import is redundant
- on line 3 that says that `java.util.List` import is redundant

--------------

**Comment by Vantuz** [comment]
I have never contributed to this project before, but I would love to start by trying to fix this myself. Do you want me to create an MR with fix or is it better for someone else to work on this?

**Comment by rnveach** [comment]
@Vantuz anyone is free to work on this once issue is approved and it is decided on correct functionality of check.

I agree that the 2 imports are redundant since the package is star imported.
@romani ping

**Comment by romani** [comment]
You are welcome with PR

**Comment by mjustin** [comment]
I see one false positive edge case with the current implementation.  Specifically, the case where you have a class with a particular name in both the package that's being wildcard imported, as well as in the current package.  In this case, having the specific import is necessary if that's the intended class, since the one in the same package take precedence over the star import.

#### Classes:
```
com.example.TestClass
com.example.ImportedClass
com.example.diffpkg.ImportedClass
com.example.diffpkg.A
com.example.diffpkg.B
```

#### com.example.TestClass.java:
```java
package com.example.*;

import com.example.diffpkg.*;
import com.example.diffpkg.ImportedClass; // If this was't here, any reference to ImportedClass would be to "com.example.ImportedClass"

public class TestClass {
  private ImportedClass importedClass;

  // ...
}
```

**Comment by romani** [comment]
>I see one false positive edge case with the current implementation.

**Comment by mjustin** [comment]
@romani To clarify: I see a problem with the code that the author provided.

**Comment by romani** [comment]
code that the author provided = code in PR ( that was closed was incomplete)

**Comment by DakshRJain737** [comment]
@romani 

> It sounds like this issue can be closed as checkstyle can not do this, or we can update doc to clearly mention this

Should'nt it be closed?

**Comment by romani** [comment]
We can detect start import and we can assume that all other import of same package are not required.
```
import java.util.*;
import java.util.ArrayList;
import java.util.List;
```

we can extend Check to do this validation, There should be special message to point user to fact that he need to make decision himself if `import java.util.*;` to be removed or single class import `import java.util.ArrayList;`

**Comment by SidheshwarSarangal** [comment]
I am on it — I will extend RedundantImportCheck to detect explicit imports covered by a wildcard import from the same package, add the dedicated violation message requested in the latest discussion, and provide regression coverage and documentation.
